src/arrayMethodJoin.js

```diff
@@ -4,8 +4,18 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      element = (element === null || element === undefined) ? '' : `${element}`;
+
+      string += (i === this.length - 1) ? element : element + separator;
```

Comment:
ternary is handy, I love it, too
but let's look at this – as you can see, you duplicate `element`, is it right? again, simple _if_ could handle it better

Reply:
also, when you 100% sure ternary is a good idea, follow this code style approach
```
 something = condition 
  ? this 
  : that;
```

Reply:
Agree, repeating is bad, changed to `if`